command/meta.go

```diff
@@ -131,23 +132,36 @@ func (m *Meta) clientConfig() *api.Config {
 		config.Namespace = m.namespace
 	}
 
-	// If we need custom TLS configuration, then set it
-	if m.caCert != "" || m.caPath != "" || m.clientCert != "" || m.clientKey != "" || m.tlsServerName != "" || m.insecure {
-		t := &api.TLSConfig{
-			CACert:        m.caCert,
-			CAPath:        m.caPath,
-			ClientCert:    m.clientCert,
-			ClientKey:     m.clientKey,
-			TLSServerName: m.tlsServerName,
-			Insecure:      m.insecure,
-		}
-		config.TLSConfig = t
-	}
-
 	if m.token != "" {
 		config.SecretID = m.token
 	}
 
+	// If the user has passed custom TLS configuration, override with that.
+	// Refactored to address issue #11539
```

Comment:
It looks like the goal of this comment is explaining the "why is this not just a single struct?" question for future readers, which is great! But this last line makes the casual reader go reference the issue in GitHub and then chase it to this PR. We can probably provide all the context we need right in-line with something like:

```suggestion
	// Override TLS configuration fields we may have received from env vars with
	// flag arguments from the user only if they're provided.
```